### ISSUE TITLE
Missing configuration directive for k8s-workload-registrar

### ISSUE THREAD

**Issue by faisal-memon** [spiffe/spire#961]
In order for the k8s-workload-regsitar to register with the spire-server shareProccessNamespace needs to be added:

```
spec:
  shareProcessNamespace: true
  containers:
    ...
```

Otherwise the spire-server will crash with the following error message:

```
level=error msg="fatal: could not resolve caller information
```

**Comment by evan2645** [comment]
Related to https://github.com/spiffe/spire/issues/934

**Comment by evan2645** [comment]
Fixed in #1013
